### PR TITLE
document and improve bazel caching args, apply to more canary jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
/area jobs
/area bazel
/kind kanarynetes
/assign @ixdy @krzyzacy

### CODE REVIEW

File: experiment/nursery/create_bazel_cache_rcs.sh [hunk 1]
```diff
@@ -20,12 +20,24 @@
...
+    # since this is the only hash our cache supports
     echo "startup --host_jvm_args=-Dbazel.DigestFunction=sha256" >> $1
+    # caching args for build
+    # - use remote strategy for all the things
     echo "build --spawn_strategy=remote" >> $1
     echo "build --strategy=Javac=remote" >> $1
     echo "build --genrule_strategy=remote" >> $1
-    echo "build --remote_rest_cache=${CACHE_HOST}" >> $1
+    # - use the http cache ...
+    echo "build --remote_http_cache=${CACHE_HOST}" >> $1
```
Comment (BenTheElder) [inline experiment/nursery/create_bazel_cache_rcs.sh]: `--remote_rest_cache` was deprecated in `0.10.0`, this is the correct flag name now.

File: experiment/nursery/create_bazel_cache_rcs.sh [hunk 2]
```diff
@@ -20,12 +20,24 @@
...
+    # - don't fail if the cache is unavailable
     echo "build --remote_local_fallback" >> $1
+    # same args for, but for test
+    echo "test --spawn_strategy=remote" >> $1
+    echo "test --strategy=Javac=remote" >> $1
+    echo "test --genrule_strategy=remote" >> $1
+    echo "test --remote_http_cache=${CACHE_HOST}" >> $1
+    echo "test --remote_local_fallback" >> $1
```
Comment (BenTheElder) [inline experiment/nursery/create_bazel_cache_rcs.sh]: AFAICT from the manual these flags weren't applied to `bazel test` previously. I think this should be enough.

**Comment by BenTheElder** [comment]
Fixed verify-config.
I've tested these flags locally against a local cache server, seems good so far.

File: experiment/nursery/create_bazel_cache_rcs.sh [hunk 3]
```diff
@@ -20,12 +20,24 @@
 CACHE_HOST="http://bazel-cache.default:8080"
 
 make_bazel_rc () {
+    # this is the default for recent releases but we set it explicitly
+    # since this is the only hash our cache supports
     echo "startup --host_jvm_args=-Dbazel.DigestFunction=sha256" >> $1
+    # caching args for build
+    # - use remote strategy for all the things
     echo "build --spawn_strategy=remote" >> $1
```
Comment (ixdy) [inline experiment/nursery/create_bazel_cache_rcs.sh]: does this mean everything is executing remotely?

File: experiment/nursery/create_bazel_cache_rcs.sh (hunk 2 above)
Comment (ixdy) [reply experiment/nursery/create_bazel_cache_rcs.sh]: are you sure? I'm pretty certain that test inherits from build.

File: experiment/nursery/create_bazel_cache_rcs.sh (hunk 3 above)
Comment (BenTheElder) [reply experiment/nursery/create_bazel_cache_rcs.sh]: no, they definitely execute locally, but this controls how sandboxing etc works as well AFAICT? this flag is from the buchgr/bazel-remote http/rest caching docs.
I'm actually about to update this to the args specified here:
https://docs.bazel.build/versions/master/remote-caching.html#read-from-and-write-to-the-remote-cache

**Comment by BenTheElder** [comment]
/hold
testing `build --experimental_remote_spawn_cache` instead of all the strategy args locally, per https://docs.bazel.build/versions/master/remote-caching.html#read-from-and-write-to-the-remote-cache which should allow sandboxed actions to be cached too

File: experiment/nursery/create_bazel_cache_rcs.sh [hunk 4]
```diff
@@ -20,11 +20,14 @@
...
 make_bazel_rc () {
+    # this is the default for recent releases but we set it explicitly
+    # since this is the only hash our cache supports
     echo "startup --host_jvm_args=-Dbazel.DigestFunction=sha256" >> $1
-    echo "build --spawn_strategy=remote" >> $1
-    echo "build --strategy=Javac=remote" >> $1
-    echo "build --genrule_strategy=remote" >> $1
-    echo "build --remote_rest_cache=${CACHE_HOST}" >> $1
+    # use remote caching for all the things
+    echo "--experimental_remote_spawn_cache" >> $1
```
Comment (BenTheElder) [inline experiment/nursery/create_bazel_cache_rcs.sh]: :( the `build` got dropped here when I switched the flags out :(
following up...

**Pull request merged**